provisioning/device/src/Transports/Mqtt/ProvisioningTransportHandlerMqtt.cs

```diff
@@ -83,21 +80,32 @@ internal ProvisioningTransportHandlerMqtt(ProvisioningClientOptions options)
 
             cancellationToken.ThrowIfCancellationRequested();
 
-            _isClosing = false;
-            _isOpening = true;
             _connectionLossCause = null;
             using IMqttClient mqttClient = s_mqttFactory.CreateMqttClient();
             MqttClientOptionsBuilder mqttClientOptionsBuilder = CreateMqttClientOptions(provisioningRequest);
             mqttClient.ApplicationMessageReceivedAsync += HandleReceivedMessageAsync;
 
+            using var connectionLostCancellationToken = new CancellationTokenSource();
+
             // Link the user-supplied cancellation token with a cancellation token that is cancelled
             // when the connection is lost so that all operations stop when either the user
             // cancels the token or when the connection is lost.
-            _connectionLostCancellationToken = new CancellationTokenSource();
             using CancellationTokenSource linkedCancellationToken =
                 CancellationTokenSource.CreateLinkedTokenSource(
                     cancellationToken,
-                    _connectionLostCancellationToken.Token);
+                    connectionLostCancellationToken.Token);
+
+            Task HandleDisconnectionAsync(MqttClientDisconnectedEventArgs disconnectedEventArgs)
+            {
+                _connectionLossCause = disconnectedEventArgs.Exception;
+
+                if (Logging.IsEnabled)
+                    Logging.Error(this, $"MQTT connection was lost '{_connectionLossCause}'.");
+
+                // If it was an unexpected disconnect. Ignore cases when the user intentionally closes the connection.
```

Comment:
What does a user intentionally closing the connection mean this in context? Do we expect HandleDisconnectionAsync to be called even when the user-supplied cancelation token expires?